src/web/frontend/components/Header.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import { Link } from "react-router-dom";
+import Rules from "./Rules";
+import '../sass/components/header.scss';
+
+type HeaderProps = {
+    onlineCount: Number
+};
+type HeaderState = {
+    showRules: Boolean,
+    showSignin: Boolean
+};
```

Comment:
It is always recommended to use `boolean`, the primitive type instead of `Boolean`. Same with `number`.
See [Typescript Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#general-types) for reference.